datadog/resource_datadog_service_level_objective.go

```diff
@@ -0,0 +1,433 @@
+package datadog
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogServiceLevelObjective() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogServiceLevelObjectiveCreate,
+		Read:   resourceDatadogServiceLevelObjectiveRead,
+		Update: resourceDatadogServiceLevelObjectiveUpdate,
+		Delete: resourceDatadogServiceLevelObjectiveDelete,
+		Exists: resourceDatadogServiceLevelObjectiveExists,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogServiceLevelObjectiveImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Common
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(val interface{}) string {
+					return strings.TrimSpace(val.(string))
+				},
+			},
+			"tags": {
+				// we use TypeSet to represent tags, paradoxically to be able to maintain them ordered;
+				// we order them explicitly in the read/create/update methods of this resource and using
+				// TypeSet makes Terraform ignore differences in order when creating a plan
+				Type:     schema.TypeSet,
```

Comment:
Interesting, the primary goal here being that you can flip the order of the tags and no diff will be shown? 💜

Reply:
yes